src/model/helpers.rs

```diff
@@ -903,27 +903,61 @@ pub struct GetNotificationsParams {
     // the notification ID to start returning messages after. Null to start with the most recent notification
     #[serde(rename = "aft")]
     pub after: Option<Uuid>,
+
+    // order unread notifications before read notifications regardless of time and paginate from there
+    #[serde(rename = "urf", default)]
+    pub unread_first: bool,
 }
 
 #[instrument(skip(postgres, metrics))]
 pub async fn get_notifications_for_subscriber(
     subscriber: Uuid,
-    GetNotificationsParams { limit, after }: GetNotificationsParams,
+    GetNotificationsParams {
+        limit,
+        after,
+        unread_first,
+    }: GetNotificationsParams,
     postgres: &PgPool,
     metrics: Option<&Metrics>,
 ) -> Result<GetNotificationsResult, sqlx::error::Error> {
-    let after_clause = if after.is_some() {
+    let after_clause = |after: Option<Uuid>, variable: &str| {
+        if after.is_some() {
+            let (unread_first_clause_left, unread_first_clause_right) = if unread_first {
+                (
+                    format!("(SELECT is_read FROM subscriber_notification WHERE id={variable}),"),
+                    "subscriber_notification.is_read,",
+                )
+            } else {
+                ("".to_owned(), "")
+            };
+            format!(
+                "
+                AND (
+                    {unread_first_clause_left}
+                    NOW() - (SELECT created_at FROM subscriber_notification WHERE id={variable}),
+                    {variable}
+                ) < (
+                    {unread_first_clause_right}
+                    NOW() - subscriber_notification.created_at,
+                    subscriber_notification.id
+                )
+                "
+            )
+        } else {
+            "".to_owned()
+        }
+    };
+    let order_by_clause = if unread_first {
         "
-        AND (
-            (SELECT created_at FROM subscriber_notification WHERE id=$3),
-            $3
-        ) > (
-            subscriber_notification.created_at,
-            subscriber_notification.id
-        )
+        subscriber_notification.is_read ASC,
+        subscriber_notification.created_at DESC,
+        subscriber_notification.id ASC
```

Comment:
Technically this changes the sort order from higher ID first to lower ID first. But this was used as a tie-breaker only when created_at was the same. This just guarantees that when paginating that there is no risk of missing or duplicate notifications and I think it makes more sense to put lower IDs first in the tiebreaker.